tests/datasets/test_sentinel.py

```diff
@@ -149,3 +155,36 @@ def test_invalid_query(self, dataset: Sentinel2) -> None:
 
     def test_float_res(self, dataset: Sentinel2) -> None:
         Sentinel2(dataset.paths, res=10.0, bands=dataset.bands)
+
+    @pytest.mark.parametrize('dataset_spec', [DatasetReader, WarpedVRT])
```

Comment:
Can't we test data that does and doesn't need reprojection instead of these dataset_spec things? It isn't clear when a real user would hit this, since the code being tested isn't something a real user would use.